### PR TITLE
native runner: convert logits to float

### PR DESCRIPTION
**Pull request by swolchok**
The native runner incorrectly assumed that the output of the model
must always be float. Now it doesn't.

### CODE REVIEW

**Comment by swolchok** [comment]
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

**Review by larryliu0820 — APPROVED** [review]

File: runner/run.cpp [hunk 1]
```diff
@@ -212,6 +212,7 @@ float* forward(Transformer* transformer, int token, int pos) {
                              .to(torch::dtype(torch::kFloat32))
                              .to(torch::kCPU);
   auto logits = result[0].data_ptr();
+  memcpy(s->logits, logits, p->vocab_size * sizeof(float));
```
Comment (larryliu0820) [inline runner/run.cpp:215]: This is for AOTI right? This is needed because the rest of the runner assumes float?

Comment (swolchok) [reply runner/run.cpp:215]: it is simply moved up from below.
